checkout/access-checkout-android/src/test/java/com/worldpay/access/checkout/validation/CVVValidatorTest.kt

```diff
@@ -1,289 +1,302 @@
     fun `given a cvv less than defaults max length then should be completely invalid and partially valid`() {
-        val cvvValidator = CVVValidatorImpl(
-            CardConfiguration(
-                emptyList(),
-                CARD_DEFAULTS
-            )
-        )
-
-        assertEquals(Pair(ValidationResult(partial = true, complete = false), null), cvvValidator.validate("12", null))
+        assertEquals(Pair(ValidationResult(partial = true, complete = false), null), cvvValidator.validate(
+            "12",
+            null,
+            CARD_CONFIG_NO_BRAND
+        ))
     }
 
     @Test
     fun `given a cvv equal to defaults max length then should be completely valid`() {
-        val cvvValidator = CVVValidatorImpl(
-            CardConfiguration(
-                emptyList(),
-                CARD_DEFAULTS
-            )
-        )
-
-        assertEquals(Pair(ValidationResult(partial = false, complete = true), null), cvvValidator.validate("1234", null))
+        assertEquals(Pair(ValidationResult(partial = false, complete = true), null), cvvValidator.validate(
+            "1234",
+            null,
+            CARD_CONFIG_NO_BRAND
+        ))
     }
 
     @Test
     fun `given a pan that matches a brand rule and a cvv with valid length then should be completely valid`() {
-        val cvvValidatorImpl = CVVValidatorImpl(CARD_CONFIG_BASIC)
-
-        assertEquals(Pair(ValidationResult(partial = false, complete = true), VISA_BRAND), cvvValidatorImpl.validate("123", VISA_PAN))
+        assertEquals(Pair(ValidationResult(partial = false, complete = true), VISA_BRAND), cvvValidator.validate(
+            "123",
+            VISA_PAN,
+            CARD_CONFIG_BASIC
+        ))
     }
 
     @Test
     fun `given a pan that matches a brand rule and a cvv with less than valid length then should be partially valid`() {
-        val cvvValidatorImpl = CVVValidatorImpl(CARD_CONFIG_BASIC)
-
-        assertEquals(Pair(ValidationResult(partial = true, complete = false), VISA_BRAND), cvvValidatorImpl.validate("1", PARTIAL_VISA))
+        assertEquals(Pair(ValidationResult(partial = true, complete = false), VISA_BRAND), cvvValidator.validate(
+            "1",
+            PARTIAL_VISA,
+            CARD_CONFIG_BASIC
+        ))
     }
 
     @Test
     fun `given a pan that matches a brand rule and a cvv with more than valid length then should be completely and partially invalid`() {
-        val cvvValidatorImpl = CVVValidatorImpl(CARD_CONFIG_BASIC)
-
-        assertEquals(Pair(ValidationResult(partial = false, complete = false), VISA_BRAND), cvvValidatorImpl.validate("12345", PARTIAL_VISA))
+        assertEquals(Pair(ValidationResult(partial = false, complete = false), VISA_BRAND), cvvValidator.validate(
+            "12345",
+            PARTIAL_VISA,
+            CARD_CONFIG_BASIC
+        ))
     }
 
     @Test
     fun `given a cvv with length on lower bounds in defaults min and max length then should be completely valid`() {
```

Comment:
I don't understand what the test does. Ignore my comment if this is will be removed/reworked anyway

Reply:
it is worded weird, not sure completely. this will be reworked too.